File: voidfindertk/popcorn/popcorn_void_finder/Example/abundance.py
import numpy as np
import matplotlib.pyplot as plt

from read_voids import read_sph, read_pop

# Reading the spherical void catalogue:
sphvoids = read_sph("sphvoids.dat")

# Reading the popcorn void catalogue:
popvoids = read_pop("popvoids.dat")

# Calculating radii histograms:
hs, edges_s = np.histogram(sphvoids['r'])
hp, edges_p = np.histogram(popvoids['reff'])

# Midpoints of bins:
mids_s = 0.5 * (edges_s[1:] + edges_s[:-1])
mids_p = 0.5 * (edges_p[1:] + edges_p[:-1])

# Plotting radii distribution:
plt.loglog(mids_s, hs, label='spherical', linewidth=2, color='blue')
plt.loglog(mids_p, hp, label='popcorn', linewidth=2, color='red')

plt.xlabel(r'$R_v$ [Mpc/h]')
plt.ylabel('Void counts')
plt.legend(loc='upper right')
plt.tight_layout()
plt.show()

File: voidfindertk/popcorn/popcorn_void_finder/Example/abundance.r
source("read_voids.r")

# Reading the spherical voids catalogue:
sphvoids <- read_sph("sphvoids.dat")

# Reading the popcorn voids catalogue:
popvoids <- read_pop("popvoids.dat")

# Getting the effective popcorn spheres:
effvoids <- pop2sph(popvoids)

# Calculating radii histograms:
hs <- hist(sphvoids$r, plot=FALSE)
hp <- hist(effvoids$r, plot=FALSE)

# Plotting radii distribution:
plot( hs$mids, hs$counts
    , log="xy"
    , type="l", lwd=2, col="blue"
    , xlab=expression(R[v]~"[Mpc/h]")
    , ylab="Void counts" 
    )

lines( hp$mids, hp$counts
     , lwd=2, col="red"
     )

legend( "topright"
      , c("spherical", "popcorn")
      , lty=1, lwd=2, col=c("blue", "red")
      , bty="n" 
      )
